packages/govuk-frontend-review/src/routes/full-page-examples.mjs

```diff
@@ -1,48 +1,81 @@
+import express from 'express'
+
 import { getFullPageExamples } from '../common/lib/files.mjs'
 import * as routes from '../views/full-page-examples/index.mjs'
 
+const router = express.Router()
 const fullPageExamples = await getFullPageExamples()
 const fullPageExampleNames = fullPageExamples.map(({ path }) => path)
 
 /**
- * @param {import('express').Application} app
+ * Handle parameter :exampleName
+ *
+ * Adds context for full page examples for `{{ example.title }}` etc
  */
-export default (app) => {
-  routes.cookieBannerEssentialCookies(app)
-  routes.cookieBannerServerSide(app)
-  routes.haveYouChangedYourName(app)
-  routes.feedback(app)
-  routes.howDoYouWantToSignIn(app)
-  routes.search(app)
-  routes.passportDetails(app)
-  routes.updateYourAccountDetails(app)
-  routes.uploadYourPhoto(app)
-  routes.uploadYourPhotoSuccess(app)
-  routes.whatIsYourAddress(app)
-  routes.whatIsYourNationality(app)
-  routes.whatIsYourPostcode(app)
-  routes.whatWasTheLastCountryYouVisited(app)
+router.param(
+  'exampleName',
 
   /**
-   * Full page examples index
+   * @param {import('express').Request} req
+   * @param {import('express').Response} res
+   * @param {import('express').NextFunction} next
+   * @param {string} exampleName
    */
-  app.get('/full-page-examples', async (req, res) => {
-    res.render('full-page-examples/index', {
-      fullPageExamples
-    })
-  })
+  (req, res, next, exampleName) => {
+    const example = fullPageExamples.find(
+      (example) => example.path === exampleName
+    )
 
-  /**
-   * Full page example
-   */
-  app.get('/full-page-examples/:exampleName', (req, res, next) => {
-    const { exampleName } = req.params
+    // Update response locals
+    res.locals.example = example
 
-    // No matching example so continue to page not found
-    if (!fullPageExampleNames.includes(exampleName)) {
-      return next()
-    }
+    next()
+  }
+)
 
-    res.render(`full-page-examples/${exampleName}/index`)
+/**
+ * Full page examples index
+ */
+router.get('/', (req, res) => {
+  res.render('full-page-examples/index', {
+    fullPageExamples
   })
+})
+
+/**
+ * Full page example 404 handler
+ */
+router.all('/:exampleName', (req, res, next) => {
+  const { exampleName } = req.params
+
+  // No matching example so flag as page not found
+  // for other routes mounted to '/:exampleName'
+  if (!fullPageExampleNames.includes(exampleName)) {
+    res.status(404)
+  }
+
+  next()
+})
```

Comment:
Good question

I added the 404 check in https://github.com/alphagov/govuk-frontend/pull/4463 via these lines:

https://github.com/alphagov/govuk-frontend/blob/12ce02a777649aa2d8cbfd79705e74eba611b4b2/packages/govuk-frontend-review/src/routes/full-page-examples.mjs#L33-L38

But it means examples with form POST bypass the `.get()` checks 🤦‍♂️

This way we can trigger a 404 and still let those custom POST routes decide what to do

Reply:
I think I'd lean towards taking advantage of the routing system for handling 404s (solution 2), as it's less code for us and takes advantage of how the routing system works. Maybe I'm biased by having relied on "let the request fall through until it hits the right route or 404s" in Rails apps routing for so long.

Wouldn't mind putting the `.post` closer to the `.param` and maybe making it a `.all` like your initial handler that was setting 404 if it works. I think the grouping and `.all` would make it read as "we handle the first part of the path this way".

Reply:
Thanks @romaricpascal

If you're comfortable with requests falling through I've pushed up 0aa225a5644813ba26dfc73bb60b3f25fdeb2b90

It's a combination of `router.all(‘/:exampleName')` with a known route check to avoid hard coding

Only downside to `route.all()` is the new `req.method !== 'GET'` call to `next()` or unhandled routes with PATCH, POST, PUT, DELETE etc will leave the server hanging

Reply:
Cheers for updating @colinrotherham.

Was terrible at explaining what I was thinking here, I was thinking of moving just the `request.post('/:exampleName', (req, res, next) => next())` right after the `param` and turning it into an `.all` as `request.all('/:exampleName', (req, res, next) => next())`. 

That way:
- we're clearly setting that the first part of the path is the name of the example as a convention in that Router.
- Express does all the heavy lifting of whether there's a route matching or not by having the routes defined later on to catch the requests when one matches, or let the request fall through to the 404 if nothing matches. 

Sorry for the lack of clarity, and happy to discuss on a quick call rather than async to wrap this one up (and likely [the other examples](https://github.com/alphagov/govuk-frontend/pull/4500) at the same time) 😊

Reply:
If it's still necessary, yeah, happy to keep it. But I'd have thought that if the `.get('/:exampleName')` is below the example's routers, the `GET` will be handled by the example's routers (as a route wouldn't call `next()`) and we want and never reach `.get('/:exampleName')`. If an example's router adds a middleware (which would call `next()`) and there's no example to render, things would fall through and 404 in the end, which makes sense as well.

But if express doesn't work this way/I misunderstood the flow of how requests fall through, very fine to keep the check for the router 😊

Reply:
Ah ha, this bit was the key feedback:

>But I'd have thought that if the `.get('/:exampleName')` is below the example's routers, the `GET` will be handled by the example's routers (as a route wouldn't call `next()`) and we want and never reach `.get('/:exampleName')`.

I've moved the custom example routers up so they match first now in 736e5c64fbce270eda0bdf6f27b13a4dc380d95d

Good to go